Source/Router.swift

```diff
@@ -98,18 +98,20 @@ public final class Router {
     private class Route: Hashable {
         let path: String
         let method: HTTPMethod
+        let queryParameters: [URLQueryItem]
         
         static func == (lhs: Router.Route, rhs: Router.Route) -> Bool {
-            return lhs.path == rhs.path && lhs.method == rhs.method
+            return lhs.path == rhs.path && lhs.method == rhs.method && lhs.queryParameters == rhs.queryParameters
         }
         
-        init(path: String, method: HTTPMethod) {
+        init(path: String, method: HTTPMethod, queryParameters: [URLQueryItem] = []) {
             self.path = path
             self.method = method
+            self.queryParameters = queryParameters
         }
         
         var hashValue: Int {
-            return path.hashValue ^ method.hashValue
+            return queryParameters.reduce(path.hashValue ^ method.hashValue) { $0.0 ^ $0.1.hashValue} << 8
```

Comment:
Not sure what you mean.. matching is done in `matchRoute(...)` here it means en extra query param will result in a different hash, so a separate Route. `/user?param=1` will not replace `/user`, but be separate. (Where we run into prioritization issues ... which route to execute when calling `www.myapi.com/user?param=1`?